analyzer/emerald/incoming.go

```diff
@@ -400,7 +432,311 @@ func extractRound(b *block.Block, txrs []*sdkClient.TransactionWithResults, logg
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "constant": true,
+      "inputs": [
+        {
+          "name": "_owner",
+          "type": "address"
+        }
+      ],
+      "name": "balanceOf",
+      "outputs": [
+        {
+          "name": "balance",
+          "type": "uint256"
+        }
+      ],
+      "payable": false,
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "constant": true,
+      "inputs": [],
+      "name": "symbol",
+      "outputs": [
+        {
+          "name": "",
+          "type": "string"
+        }
+      ],
+      "payable": false,
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "constant": false,
+      "inputs": [
+        {
+          "name": "_to",
+          "type": "address"
+        },
+        {
+          "name": "_value",
+          "type": "uint256"
+        }
+      ],
+      "name": "transfer",
+      "outputs": [
+        {
+          "name": "success",
+          "type": "bool"
+        }
+      ],
+      "payable": false,
+      "stateMutability": "nonpayable",
+      "type": "function"
+    },
+    {
+      "constant": true,
+      "inputs": [
+        {
+          "name": "_owner",
+          "type": "address"
+        },
+        {
+          "name": "_spender",
+          "type": "address"
+        }
+      ],
+      "name": "allowance",
+      "outputs": [
+        {
+          "name": "remaining",
+          "type": "uint256"
+        }
+      ],
+      "payable": false,
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "name": "_from",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "name": "_to",
+          "type": "address"
+        },
+        {
+          "indexed": false,
+          "name": "_value",
+          "type": "uint256"
+        }
+      ],
+      "name": "Transfer",
+      "type": "event"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "name": "_owner",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "name": "_spender",
+          "type": "address"
+        },
+        {
+          "indexed": false,
+          "name": "_value",
+          "type": "uint256"
+        }
+      ],
+      "name": "Approval",
+      "type": "event"
+    }
+  ]
+`))
+	if err != nil {
+		panic(err)
+	}
+}
+
+// nolint: unparam
+func evmCallWithABI(
+	ctx context.Context,
+	client *oasis.RuntimeClient,
+	round uint64,
+	contractEthAddr []byte,
+	contractABI *abi.ABI,
+	name string,
+	outI interface{},
+	in ...interface{},
+) error {
+	// https://github.com/oasisprotocol/emerald-web3-gateway/blob/v3.0.0/rpc/eth/api.go#L408
+	gasPrice := []byte{1}
+	gasLimit := uint64(30_000_000)
+	caller := ethCommon.Address{1}.Bytes()
```

Comment:
using the same caller that the web3 gateway uses so that we don't have to decide for ourselves https://github.com/oasisprotocol/emerald-web3-gateway/blob/v3.0.0/rpc/eth/api.go#LL405

maybe we don't want it to act as unauthenticated? dunno

Reply:
we'll be going on to use the same settings as the web3 gateway. thanks for asking